### PR TITLE
support building gerrit and plugins

### PR DESCRIPTION
**Pull request by mulby**
This is what I used to build the artifacts that are currently being used to deploy gerrit.  I ran this playbook on a bare metal ubuntu precise64 vagrant image.

### CODE REVIEW

**Comment by mulby** [comment]
@gwprice @e0d @cpennington @wedaly 

Not sure if ansible is the right "home" for this automation, open to suggestions.  Also, I suppose we should probably fork the repos I modified, apply the patches to the forks and then clone those instead of the current clone & patch method.

**Comment by hkim823** [comment]
@mulby this is pretty old, was this solved in another way since opening?

**Comment by mulby** [comment]
Nope. This is the playbook I used to build gerrit and its associated
plugins. Just never got merged in.

File: playbooks/edx-east/gerrit_build.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+- name: Build gerrit
+  hosts: gerrit_build
+  sudo: True
+  gather_facts: True
+  vars_files:
+    - "{{ secure_dir }}/vars/gerrit.yml"
+  pre_tasks:
+    - name: playbook pre-requisites
+      apt: pkg=curl update_cache=yes
```
Comment (jarv) [inline playbooks/edx-east/gerrit_build.yml:15]: can we move this package into the role if it's required?

File: playbooks/roles/buck/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# Defaults for role buck
+# 
+
+buck_debian_pkgs:
+  - git
+  - ant
+  - curl
+  - zip
+
+BUCK_HOME: /opt/buck
```
Comment (jarv) [inline playbooks/roles/buck/defaults/main.yml:20]: nit: put the all caps vars first

Comment (jarv) [inline playbooks/roles/buck/defaults/main.yml:20]: I would rather see this role depend on common if it's not too much overhead and have it installed in the common app dir. (/edx/app by default)

File: playbooks/roles/gerrit_github_api_source/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#
+##
+# Defaults for role gerrit_github_api_source
+# 
+
+gerrit_github_api_source_debian_pkgs:
+  - git
+  - maven
+
+GERRIT_GITHUB_API_SOURCE_HOME: /opt/gerrit-github-api-source
```
Comment (jarv) [inline playbooks/roles/gerrit_github_api_source/defaults/main.yml:18]: also would like to this to depend on common and installed in the common app dir if possible

File: playbooks/roles/gerrit_github_api_source/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# Dependencies: None
+#
+# 
+
+- name: gerrit_github_api_source | system packages
+  apt: pkg={{ item }}
+  with_items: gerrit_github_api_source_debian_pkgs
+
+- name: gerrit_github_api_source | source checked out
+  git: repo={{ gerrit_github_api_source_repo }} dest={{ GERRIT_GITHUB_API_SOURCE_HOME }} version={{ gerrit_github_api_source_version }}
```
Comment (jarv) [inline playbooks/roles/gerrit_github_api_source/tasks/main.yml:26]: nit: break this into multiple lines

**Comment by e0d** [comment]
@mulby we're cleaning house; should this be closed?

**Comment by mulby** [comment]
sure? I guess I can always revive it if/when we start using gerrit more broadly

**Pull request closed without merging**